c2rust-analyze/src/c_void_casts.rs

```diff
@@ -347,14 +362,28 @@ impl<'tcx> CVoidCasts<'tcx> {
                     To => args[0].place().unwrap(),
                 };
                 let c_void_ptr = CVoidPtr::checked(c_void_ptr, &body.local_decls, tcx);
-                let get_cast =
-                    move |stmt: &Statement<'tcx>| c_void_ptr.get_cast_from_stmt(direction, stmt);
+                let get_cast = move |(idx, stmt): (usize, &Statement<'tcx>)| {
```

Comment:
@kkysen I don't really like this, and it was a function of needing type annotations for this closure in `find_last_cast_curr_block` for example. I'm open to cleaner ways of going about this.